src/target/riscv/riscv.h

```diff
@@ -152,6 +152,17 @@ struct riscv_info {
 	/* record the tinfo of each trigger */
 	unsigned int trigger_tinfo[RISCV_MAX_TRIGGERS];
 
+	/* record the tinfo_version of each trigger */
+	int64_t trigger_tinfo_version[RISCV_MAX_TRIGGERS];
```

Comment:
`int64_t` looks excessive here to store a version which is no longer than 8 bits.

Please, use just an `int`.

Also `-1` can be defined as "unknown tinfo.version". See my other comment.

Reply:
addressed